modules/@apostrophecms/modal/ui/apos/components/AposModal.vue

```diff
@@ -187,12 +197,19 @@ export default {
     }
   },
   watch: {
-    modalReady (newVal) {
-      this.$nextTick(() => {
-        if (newVal && this.modal.trapFocus && this.$refs.modalEl) {
-          this.trapFocus();
-        }
-      });
+    // Simple way to re-trigger focusable elements
+    // that might have been created or removed
+    // after an update, like an XHR call to get the
+    // pieces list in the AposDocsManager modal, for instance.
+    triggerFocusRefresh (newVal) {
+      if (this.shouldTrapFocus) {
+        this.$nextTick(this.trapFocus);
+      }
+    }
```

Comment:
this is the trick to refresh new elements to focus after some update that happened inside a parent component (AposDocEditor, AposDocsManager, AposPagesManager...)